aggregator_api/src/lib.rs

```diff
@@ -351,86 +384,147 @@ mod models {
         pub(crate) aggregator_auth_token: Option<String>,
-        pub(crate) collector_auth_token: Option<String>,
     }
 
     #[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
     pub(crate) struct TaskResp {
+        /// ID of the DAP Task.
         pub(crate) task_id: TaskId,
-        pub(crate) leader_endpoint: Url,
-        pub(crate) helper_endpoint: Url,
+        /// URL relative to which this task's peer aggregator's DAP API can be found. The peer
+        /// aggregator plays the DAP role opposite to the one in the `role` field.
+        pub(crate) peer_aggregator_endpoint: Url,
+        /// DAP query type for this task.
         pub(crate) query_type: QueryType,
+        /// The VDAF being run by this task.
         pub(crate) vdaf: VdafInstance,
+        /// The role that this aggregator will play in this task.
         pub(crate) role: Role,
-        pub(crate) vdaf_verify_keys: Vec<String>,
+        /// The VDAF verification key used for this DAP task, as Base64 encoded bytes. Task ID is
+        /// derived from the verify key.
+        pub(crate) vdaf_verify_key: String,
+        /// The maximum number of times a given batch may be collected.
         pub(crate) max_batch_query_count: u64,
+        /// The time after which the task is considered invalid.
         pub(crate) task_expiration: Option<Time>,
+        /// The age after which a report is considered to be "expired" and will be considered a
+        /// candidate for garbage collection.
         pub(crate) report_expiry_age: Option<Duration>,
+        /// The minimum number of reports in a batch to allow it to be collected.
         pub(crate) min_batch_size: u64,
+        /// The duration to which clients should round their reported timestamps.
         pub(crate) time_precision: Duration,
+        /// How much clock skew to allow between client and aggregator. Reports from
+        /// farther than this duration into the future will be rejected.
         pub(crate) tolerable_clock_skew: Duration,
+        /// The authentication token for inter-aggregator communication in this task, as Base64
+        /// encoded bytes.
+        /// If `role` is Leader, this token is used by the aggregator to authenticate requests to
+        /// the Helper. If `role` is Helper, this token is used by the aggregator to authenticate
+        /// requests from the Leader.
+        // TODO(#1509): This field will have to change as Janus helpers will only store a salted
+        // hash of aggregator auth tokens.
+        pub(crate) aggregator_auth_token: String,
+        /// The authentication token used by the task's Collector to authenticate to the Leader, as
+        /// Base64 encoded bytes.
+        /// `Some` if `role` is Leader, `None` otherwise.
+        // TODO(#1509) This field will have to change as Janus leaders will only store a salted hash
+        // of collector auth tokens.
+        pub(crate) collector_auth_token: Option<String>,
```

Comment:
Where is it intended for the subscriber to get the collector auth token to run a collector?

Reply:
See [the sequence diagram](https://private-user-images.githubusercontent.com/19212995/244223772-4e0db56a-f55c-48de-9b09-dd15c0bed3b4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJrZXkiOiJrZXkxIiwiZXhwIjoxNjg3MzcxNjc5LCJuYmYiOjE2ODczNzEzNzksInBhdGgiOiIvMTkyMTI5OTUvMjQ0MjIzNzcyLTRlMGRiNTZhLWY1NWMtNDhkZS05YjA5LWRkMTVjMGJlZDNiNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMwNjIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMDYyMVQxODE2MTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lYjQwNDUyMTVkMWUzY2Q3NmIyMzRiZjU4NjJhY2I2ZTYyOTdlNWI4NDE0MzZkYWQwOTI4MDgwYzRhYjA1ZWIzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ngM2dTLOSF0ZuUDdLI9r6GWn34Qp2qTRWUYy_9OXC8c) in #1486. Short version:

- When the Leader is provisioning the task into itself, it generates the collector auth token and stores it in its own datastore.
- Leader delivers the collector auth token to divviup-api in this `TaskResp` message
- divviup-api then provides the collector auth token to the subscriber in the response to the `NewTask` message
- Collector is expected to store the auth token and use it later to authenticate collection job requests to the leader

The nuance here is that we need to give the subscriber the collector auth token "in the clear" when the task is created. But when we do #1509, Janus will start storing a token hash instead of the token. That means that a `GET /tasks/:id` request cannot contain the actual auth token, because the leader doesn't have it anymore at that point. I guess at that point this field can just be `None`, so maybe no message field changes will be needed.